### PR TITLE
chore(docs): change the CI references in our contributing docs and add missing sections

### PR DESCRIPTION
**Pull request by JGAntunes**
🎉 Thanks for submitting a pull request! 🎉

#### Summary

Part of https://github.com/netlify/pod-workflow/issues/251. This updates our `CONTRIBUTING.md` with the new CI details. As a bonus also adds a couple of useful sections. Tagged you @KyleBlankRollins in case you want/have some time to review. Despite not being customer facing docs it's contributor facing docs and thought you could lend a hand 😄 

---

For us to review and ship your PR efficiently, please perform the following steps:

- [x] Open a [bug/issue](https://github.com/netlify/build-image/issues/new/choose) before writing your code 🧑‍💻. This ensures we can discuss the changes and get feedback from everyone that should be involved. If you\`re fixing a typo or something that\`s on fire 🔥 (e.g. incident related), you can skip this step.
- [x] Read the [contribution guidelines](../CONTRIBUTING.md) 📖. This ensures your code follows our style guide and passes our tests.
- [x] Update or add tests (if any source code was changed or added) 🧪
- [x] Update the [included software doc](../included_software.md) (if you updated included software) 📄
- [x] Update or add documentation (if features were changed or added) 📝
- [x] Make sure the status checks below are successful ✅

**A picture of a cute animal (not mandatory, but encouraged)**

### CODE REVIEW

**Review by ehmicky — DISMISSED** [review]
Nice!

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -31,6 +25,33 @@ Some examples:
 - `make run` build the base image and run a bash shell in a container based in it in interactive mode.
 - `make run-local` volume the build scripts and run a bash shell in a container based in the build image.
 
+Our [changelog](./CHANGELOG.md) is powered by an automated [release-please](https://github.com/googleapis/release-please) action, which relies on commits following the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0-beta.2/#summary).
+So make sure that **both your PRs and commmits follow the conventional commits format** for the CI validation to succeed.
```
Comment (KyleBlankRollins) [inline CONTRIBUTING.md]: [pebble] Minor edit for concision and clarity. Moved "CI validation to succeed" Starting with the "why" here should help folks understand what comes after better.
```suggestion
For the CI validation to succeed, make sure that **your PRs and commits follow the conventional commits format**.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -31,6 +25,33 @@ Some examples:
...
+
+### CI
+
+The bulk of our CI work takes place in [Circle CI](https://app.circleci.com/pipelines/github/netlify/build-image).
+
+If **you're part of the Netlify org** and have write access to the repo our pipeline will take care of:
```
Comment (KyleBlankRollins) [inline CONTRIBUTING.md]: [dust] Comma!
```suggestion
If **you're part of the Netlify org** and have write access to the repo, our pipeline will take care of:
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -31,6 +25,33 @@ Some examples:
...
+- Linting the Dockerfile
+- Build your Dockerfile, tag it and push it to our [docker hub repo](https://hub.docker.com/r/netlify/build)
```
Comment (KyleBlankRollins) [inline CONTRIBUTING.md]: [dust] Serial (Oxford) comma!
```suggestion
- Build your Dockerfile, tag it, and push it to our [docker hub repo](https://hub.docker.com/r/netlify/build)
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -31,6 +25,33 @@ Some examples:
...
+- Run the automated [bats tests](#tests)
+
+
+If you **don't have access writes to the repo** and are submitting a PR via a forked repo, the CI pipeline will still execute for you. The main difference is that it won't push your built image to our docker repo.
```
Comment (KyleBlankRollins) [inline CONTRIBUTING.md]: [pebble] Swap "access" and "writes" order. This mimics what we say on line 35.
```suggestion
If you **don't have write access to the repo** and are submitting a PR via a forked repo, the CI pipeline will still execute for you. The main difference is that it won't push your built image to our docker repo.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -31,6 +25,33 @@ Some examples:
...
+If required, someone with access writes to the repo can trigger the push for you, so if you require it (mainly useful for testing purposes) reach out to someone in the team :+1:
```
Comment (KyleBlankRollins) [inline CONTRIBUTING.md]: [pebble] Same as previous edit. And a couple minor edits for clarity.
```suggestion
If required, someone with write accesss to the repo can trigger the push for you. If you require it (mainly useful for testing purposes) reach out to someone on the team :+1:
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -31,6 +25,33 @@ Some examples:
...
+
+## Releasing
+
+1. Merge the relevant package release PR created by `release-please`
+2. Wait for the CI pipelines to finish
+3. Create a PR in the [buildbot](https://github.com/netlify/buildbot) to bump [the version](https://github.com/netlify/buildbot/blob/0ada244ab84a1759a70d6b2cfc27c9987b5c77ca/.circleci/config.yml#L141-L150) of the `build-image`.
+
+### Running Test `buildbot` Releases
+
+If you want to test a particular `build-image` change before going through the regular release process you can do so by creating a PR following the process above :point-up: and pointing to any `build-image` you want. Any branch
```
Comment (KyleBlankRollins) [inline CONTRIBUTING.md]: [dust] Another comma!
```suggestion
If you want to test a particular `build-image` change before going through the regular release process, you can do so by creating a PR following the process above :point-up: and pointing to any `build-image` you want. Any branch
```

Comment (KyleBlankRollins) [inline CONTRIBUTING.md]: Is there an unintentional line break after "Any branch"?

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -31,6 +25,33 @@ Some examples:
...
+creates a valid `build-image` tag in the docker registry. Once the PR in `buildbot` is created, refer to `buildbot`'s [test instructions](https://github.com/netlify/buildbot#testing-builds-on-a-live-test-site) in order to test your particular change.
```
Comment (KyleBlankRollins) [inline CONTRIBUTING.md]: [pebble] Edited for clarity. Let me know if I got anything wrong here.
```suggestion
creates a valid `build-image` tag in the docker registry. Once the PR in `buildbot` is created, you can test your `build-image` change by referring to `buildbot`'s [test instructions](https://github.com/netlify/buildbot#testing-builds-on-a-live-test-site).
```

**Review by KyleBlankRollins — DISMISSED** [review]
Thanks for tagging me and making the contributing doc better, @JGAntunes!

I have some in-line comments for commas and minor edits for concision and clarity. None of them should block you from merging, but should benefit readers if you take the suggested changes.

File: CONTRIBUTING.md (hunk 6 above)
Comment (JGAntunes) [reply CONTRIBUTING.md]: Not really I believe 🤔 but it does not impact the final document - https://github.com/netlify/build-image/blob/chore/update-contributing-ci/CONTRIBUTING.md

**Comment by JGAntunes** [comment]
Thanks for the review @KyleBlankRollins 🙏 Will need another 👍 though CC @ehmicky @KyleBlankRollins?

**Review by ehmicky — APPROVED** [review]

**Pull request merged**